src/test/java/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,132 @@
+                "Effective Java", "Joshua Bloch", "00-000-000-30",
+                new BigDecimal("110.00"), "some description",
+                "https://some-image-url.com", List.of(1L));
+
+        MvcResult mvcResult = mockMvc.perform(post("/books")
+                        .content(objectMapper.writeValueAsString(bookRequestDto))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(
+                mvcResult.getResponse().getContentAsString(), BookDto.class);
+
+        assertThat(actual)
+                .hasFieldOrPropertyWithValue("id", 3L)
+                .hasFieldOrPropertyWithValue("title", bookRequestDto.title());
+    }
+
+    @Test
+    @DisplayName("Get book by valid id")
+    @WithMockUser(username = "admin@test.com", roles = {"USER", "ADMIN"})
+    void getBook_byValidId_returnsDto() throws Exception {
+        Long bookId = 1L;
+
+        BookDto bookDto = new BookDto();
+        bookDto.setId(bookId);
+        bookDto.setTitle("Clean Code");
+        bookDto.setAuthor("Robert Martin");
+        bookDto.setIsbn("00-000-000-01");
+        bookDto.setPrice(BigDecimal.valueOf(100.45));
+
+        MvcResult mvcResult = mockMvc.perform(get("/books/" + bookId))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actual = objectMapper.readValue(mvcResult.getResponse()
+                .getContentAsString(), BookDto.class);
+
+        assertThat(actual).isNotNull()
+                .hasFieldOrPropertyWithValue("id", bookId)
+                .hasFieldOrPropertyWithValue("title", bookDto.getTitle())
+                .hasFieldOrPropertyWithValue("author", bookDto.getAuthor())
+                .hasFieldOrPropertyWithValue("isbn", bookDto.getIsbn())
+                .hasFieldOrPropertyWithValue("price", bookDto.getPrice());
+    }
+
+    @Test
+    @DisplayName("Get book by incorrect id")
+    @WithMockUser(username = "admin@test.com", roles = {"USER", "ADMIN"})
+    void getBook_byIncorrectId_throwsException() throws Exception {
+        Long bookId = 10L;
+
+        MvcResult mvcResult = mockMvc.perform(get("/books/" + bookId))
+                .andExpect(status().isNotFound())
+                .andReturn();
+
+        Exception resolvedException = mvcResult.getResolvedException();
+        Assertions.assertNotNull(resolvedException);
+        Assertions.assertTrue(resolvedException instanceof EntityNotFoundException);
+    }
+
+    @Test
+    @DisplayName("Delete book by valid id")
+    @WithMockUser(username = "admin", roles = {"USER", "ADMIN"})
+    void deleteBook_byValidId() throws Exception {
+        Long bookId = 1L;
+
+        mockMvc.perform(delete("/books/" + bookId))
+                .andExpect(status().isNoContent())
+                .andReturn();
+
+        assertThat(bookRepository.findById(bookId)).isEmpty();
```

Comment:
it is not really  good to use the method  of the repository  in your test method